### PR TITLE
Optimize Array.info and Group.info

### PR DESCRIPTION
**Pull request by dcherian**
Avoid repeated computes of the same value `getsize`

From 6 getsize requests to 1.

TODO:
* [ ] Add unit tests and/or doctests in docstrings
* [x] Changes documented in docs/release.rst
* [x] GitHub Actions have all passed
* [x] Test coverage is 100% (Codecov passes)

### CODE REVIEW

File: zarr/util.py [hunk 1]
```diff
@@ -408,14 +408,13 @@ def info_html_report(items) -> str:
 class InfoReporter:
     def __init__(self, obj):
         self.obj = obj
+        self.items = self.obj.info_items()
 
     def __repr__(self):
-        items = self.obj.info_items()
-        return info_text_report(items)
+        return info_text_report(self.items)
```
Comment (dcherian) [inline zarr/util.py:414]: The previous version called `info_items` twice per repr in a notebook.

File: zarr/core.py [hunk 2]
```diff
@@ -176,7 +176,7 @@ def __init__(
         )
 
         # initialize info reporter
-        self._info_reporter = InfoReporter(self)
+        # self._info_reporter = InfoReporter(self)
```
Comment (jhamman) [inline zarr/core.py:179]:
```suggestion
```

File: zarr/hierarchy.py [hunk 3]
```diff
@@ -211,7 +211,7 @@ def __init__(
         )
 
         # setup info
-        self._info = InfoReporter(self)
+        # self._info = InfoReporter(self)
```
Comment (jhamman) [inline zarr/hierarchy.py:214]:
```suggestion
```

**Review by jhamman — APPROVED** [review]
@dcherian - brilliant! 

Makes me wonder how many of these patterns exist in our zarr-Python today! ☠️

File: docs/release.rst [hunk 4]
```diff
@@ -21,6 +21,8 @@ Unreleased
 Enhancements
 ~~~~~~~~~~~~
 
+* Optimize ``Array.info``.
```
Comment (dcherian) [inline docs/release.rst:24]:
```suggestion
* Optimize ``Array.info`` so that it calls `getsize` only once.
```

**Pull request merged**